src/types/observablearray.ts

```diff
@@ -124,8 +133,14 @@ class ObservableArrayAdministration<T> implements IInterceptable<IArrayWillChang
 		if (oldLength !== this.lastKnownLength)
 			throw new Error("[mobx] Modification exception: the internal structure of an observable array was changed. Did you use peek() to change it?");
 		this.lastKnownLength += delta;
-		if (delta > 0 && oldLength + delta > OBSERVABLE_ARRAY_BUFFER_SIZE)
-			reserveArrayBuffer(oldLength + delta);
+		if (delta > 0 && oldLength + delta + 1 > OBSERVABLE_ARRAY_BUFFER_SIZE)
+			reserveArrayBuffer(oldLength + delta + 1);
+		if (delta > 0)
```

Comment:
It may be useful for quick publishing some minor fixes. My npm: https://www.npmjs.com/~andykog